### PR TITLE
[plugin.video.worldstarhiphop] 1.0.13+matrix.1

### PR DESCRIPTION
**Pull request by skipmodea1**
### Description
Made a matrix version now that youtube.dl is available in matrix.
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.worldstarhiphop/addon.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import xbmc
+
+from resources.lib.worldstarhiphop_const import ADDON, DATE, VERSION, BASEURL
+
+# Parse parameters...
+if len(sys.argv[2]) == 0:
+    #
+    # Main menu
+    #
+    xbmc.log("[ADDON] %s, Python Version %s" % (ADDON, str(sys.version)), xbmc.LOGDEBUG)
```
Comment (enen92) [inline plugin.video.worldstarhiphop/addon.py]:
```suggestion
    xbmc.log(f"[ADDON] {ADDON}, Python Version {sys.version}", xbmc.LOGDEBUG)
```

You might want to have a look into python3 formatted strings in the future

File: plugin.video.worldstarhiphop/addon.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    <import addon="script.module.future"            version="0.0.1"/>
+  	<import addon="script.module.html5lib"          version="0.999.0"/>
+	<import addon="script.module.youtube.dl"        version="20.616.0+matrix.1"/>
+  </requires>
+	<extension point="xbmc.python.pluginsource" library="addon.py">
+		<provides>video</provides>
+	</extension>
+  <extension point="xbmc.addon.metadata">
+	<platform>all</platform>
+	<summary lang="en">Worldstarhiphop is home to everything entertainment and hip hop</summary>
```
Comment (enen92) [inline plugin.video.worldstarhiphop/addon.xml]: en_GB

File: plugin.video.worldstarhiphop/addon.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    <description lang="en">Worldstarhiphop is home to everything entertainment and hip hop</description>
```
Comment (enen92) [inline plugin.video.worldstarhiphop/addon.xml]: en_GB

File: plugin.video.worldstarhiphop/addon.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    <disclaimer lang="en">For bugs, requests or general questions visit the Worldstarhiphop thread on the Kodi forum</disclaimer>
```
Comment (enen92) [inline plugin.video.worldstarhiphop/addon.xml]: en_GB

File: plugin.video.worldstarhiphop/addon.xml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <platform>all</platform>
+	<license>GPL-2.0-or-later</license>
+    <forum>http://forum.kodi.tv/showthread.php?tid=229385</forum>
+    <website>http://worldstarhiphop.com/</website>
+    <source>https://github.com/skipmodea1/plugin.video.worldstarhiphop.python3</source>
+    <assets>
+      <icon>resources/icon.png</icon>
+      <fanart>resources/fanart.jpg</fanart>
+    </assets>
+    <news>v1.0.13 (2030-10-25)
```
Comment (enen92) [inline plugin.video.worldstarhiphop/addon.xml]: Back to the future? :)

File: plugin.video.worldstarhiphop/changelog.txt [hunk 6]
```diff
@@ -0,0 +1,51 @@
+v1.0.13 (2030-10-25)
```
Comment (enen92) [inline plugin.video.worldstarhiphop/changelog.txt]: here too :)

File: plugin.video.worldstarhiphop/resources/lib/worldstarhiphop_play.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+                        video_url = video_urls[0]['src']
+
+                        log("video_url", video_url)
+
+                        # make youtube plugin url
+                        pos_of_last_question_mark = video_url.rfind("?")
+                        video_url = video_url[0: pos_of_last_question_mark]
+                        video_url_len = len(video_url)
+                        youtubeID = video_url[len("http://www.youtube.com/embed/"):video_url_len]
+                        youtube_url = 'plugin://plugin.video.youtube/play/?video_id=%s' % youtubeID
```
Comment (enen92) [inline plugin.video.worldstarhiphop/resources/lib/worldstarhiphop_play.py:142]: you might want to support tubed too

File: plugin.video.worldstarhiphop/resources/settings.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (enen92) [inline plugin.video.worldstarhiphop/resources/settings.xml]: You might want to convert this to the new settings format. There's an addon you can use to do that (not mandatory): https://gitlab.com/ronie/script.settings-convert

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments. Addon.xml is mandatory, the rest is optional

File: plugin.video.worldstarhiphop/resources/lib/worldstarhiphop_play.py (hunk 7 above)
Comment (skipmodea1) [reply plugin.video.worldstarhiphop/resources/lib/worldstarhiphop_play.py:142]: What do u mean with this?

Comment (enen92) [reply plugin.video.worldstarhiphop/resources/lib/worldstarhiphop_play.py:142]: Tubed is a new YouTube add-on available for matrix: https://forum.kodi.tv/showthread.php?tid=358806

Comment (skipmodea1) [reply plugin.video.worldstarhiphop/resources/lib/worldstarhiphop_play.py:142]: I'll look into it in the future.

File: plugin.video.worldstarhiphop/addon.py (hunk 1 above)
Comment (skipmodea1) [reply plugin.video.worldstarhiphop/addon.py]: Done

File: plugin.video.worldstarhiphop/addon.xml (hunk 2 above)
Comment (skipmodea1) [reply plugin.video.worldstarhiphop/addon.xml]: Done

File: plugin.video.worldstarhiphop/addon.xml (hunk 3 above)
Comment (skipmodea1) [reply plugin.video.worldstarhiphop/addon.xml]: Done

File: plugin.video.worldstarhiphop/addon.xml (hunk 4 above)
Comment (skipmodea1) [reply plugin.video.worldstarhiphop/addon.xml]: Done

File: plugin.video.worldstarhiphop/addon.xml (hunk 5 above)
Comment (skipmodea1) [reply plugin.video.worldstarhiphop/addon.xml]: Done

File: plugin.video.worldstarhiphop/changelog.txt (hunk 6 above)
Comment (skipmodea1) [reply plugin.video.worldstarhiphop/changelog.txt]: Done

File: plugin.video.worldstarhiphop/resources/settings.xml (hunk 8 above)
Comment (skipmodea1) [reply plugin.video.worldstarhiphop/resources/settings.xml]: Done

**Review by enen92 — APPROVED** [review]

**Pull request merged**